packages/statemanager/src/cache/originalStorageCache.ts

```diff
@@ -4,6 +4,17 @@ import type { Address } from '@ethereumjs/util'
 
 type getContractStorage = (address: Address, key: Uint8Array) => Promise<Uint8Array>
 
+/**
+ * Helper class to cache original storage values (so values already being present in
+ * the pre-state of a call), mainly for correct gas cost calculation in EVM/VM.
+ *
+ * TODO: Usage of this class is very implicit through the injected `getContractStorage()`
+ * method bound to the calling state manager. It should be examined if there are alternative
+ * designs being more transparent and direct along the next breaking release round.
+ *
+ * @deprecated This helper class has been moved to the @ethereumjs/common package
+ * and will be removed from this package along the next breaking release round.
```

Comment:
Can't we technically remove this here by re-exporting this `OriginalStorageCache` from common? :thinking:

Reply:
It's not clear if people are importing this from `statemanager` and things would otherwise break for them. Therefore only the deprecation for now.

This might not be a thing anyhow depending on how we decide on where to place the whole thing.